packages/examples/bike_simulator.py

```diff
@@ -38,9 +38,14 @@
 
 class PackageClass(LiotaPackage):
 
-    def run(self, registry):
+    def run(self, registry, package_record):
         from liota.entities.devices.bike_simulated import BikeSimulated
         import pint
+        from liota.lib.utilities.utility import check_integrity
+
+        # verify file integrity first
+        if (check_integrity(package_record, __file__) == False):
```

Comment:
No, your mind is restricted to package manager. Let me provide you some examples of how sha1 checksum work for downloading a file by user and receiving a package from network interface.
1. The file provider will indicate SHa1 number in downloading operation, and when user finishes downloading, they need a tool to calculate sha1 checksum and compare it with that known checksum.
2. before TXer sends a package, then calculate checksum and insert it into the package; the receiver will extract that checksum, calculate checksum with the same way for the received package, and compare whether 2 checksums are the same.

Here for our package manager, it first gets loading cmd from user, and will do sanity check and figure out which file it will really load. Based on final filename, package manager can then calculate SHA1. 

Now, package manager will call the Liota Package's run() to execute, at that point, it should provide the SHA1 for the user to verify whether I am running the file that package manager is asking me to run.